### PR TITLE
#2116 Add missing menus on SemanticBrowser and Interpreter

### PR DESCRIPTION
**Pull request by georgiana-ecobici**
The menu shall be enabled on Elements and DRepresentationDescriptor.

Change-Id: Ibfe9112560d7bf24a544373d2c4482e67eda79b6
Signed-off-by: Georgiana Ecobici <georgiana-elena.ecobici@thalesgroup.com>

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/plugin.xml [hunk 1]
```diff
@@ -255,10 +267,33 @@
...
+                               value="org.eclipse.emf.ecore.EObject">
+                           </instanceof>
+                           <adapt
+                               type="org.polarsys.kitalpha.emde.model.Element">
+                           </adapt>
+                           <adapt
+                               type="org.eclipse.sirius.viewpoint.DRepresentationDescriptor">
+                           </adapt>
+                      </or>
+                      <reference
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/plugin.xml:289]: For this one reference, I think it's  for the Diffmerge view.
Can you try to compare model and see if your new Delete menu appear in the popup of an element ?
I think it shall not appear as there is probably another "delete"

**Comment by pdulth** [comment]
Hi, 
I have tried a bit, 
If you display Root Logical Function in semantic browser,  when clicking on Delete on it, it is possible to delete it.
whereas in project explorer, if you try to delete the Root Logical Function, there is an error message preventing its deletion.
(this element is "protected" and can't be deleted.
Can you look for it ?
Thanks

**Comment by pdulth** [comment]
The "Delete keyboard" is not enabled on it also

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/plugin.xml (hunk 1 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/plugin.xml]: Yes, I'll check.

**Comment by georgiana-ecobici** [comment]
> 
> 
> The "Delete keyboard" is not enabled on it also

Ok, I'll test.

**Comment by georgiana-ecobici** [comment]
> 
> 
> Hi,
> I have tried a bit,
> If you display Root Logical Function in semantic browser, when clicking on Delete on it, it is possible to delete it.
> whereas in project explorer, if you try to delete the Root Logical Function, there is an error message preventing its deletion.
> (this element is "protected" and can't be deleted.
> Can you look for it ?
> Thanks

Ok, Thanks. I'll check.

**Comment by pdulth** [comment]
> 
> 
> > Hi,
> > I have tried a bit,
> > If you display Root Logical Function in semantic browser, when clicking on Delete on it, it is possible to delete it.
> > whereas in project explorer, if you try to delete the Root Logical Function, there is an error message preventing its deletion.
> > (this element is "protected" and can't be deleted.
> > Can you look for it ?
> > Thanks
> 
> Ok, Thanks. I'll check.

There is a method "setPreventProtectedElementsDeletion on DeleteCommand
https://github.com/eclipse/capella/blob/85be3c77c67169fd463cee64e64cf243f97f1d7d/core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/CapellaDeleteCommand.java#L106

probably enable it on you menu. 
Idk why its false by default on constructor of CapellaDeleteCommand but seems too impacting to enable it on 5.1.0

**Review by pdulth — APPROVED** [review]

**Pull request merged**